### PR TITLE
throw exception when multiple format have the same key

### LINKED ISSUE #3038: Sulu internal format avoid cropping for new format
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.4

#### Actual Behavior

when create the following format its not available in the crop:

#### Expected Behavior

When the user define an own 400x400 it should be available in cropping so internal should be false there.

#### Steps to Reproduce

Create a format in `app/config/image-formats.xml`:

```xml
    <format>
        <name>400x400</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">400</parameter>
                    <parameter name="y">400</parameter>
                </parameters>
            </command>
        </commands>
    </format>
```

its not available in the cropping dropdown because I think it is still marked as internal.

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #3038 
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR checks if a format key is used multiple times, and throws an exception in case it does.

#### To Do

- [x] Update sulu-minimal
- [x] Update sulu-standard

### CODE REVIEW

**Comment by danrot** [comment]
I would love to write unit tests for this, but this and the other PR for the  SuluThemeBundle are relying on some files to exist. Tests could be written if we would do something like Symfony did with the [ClockMock](https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/PhpUnit/ClockMock.php), but I am not sure where we should put that 😕

**Comment by chirimoya** [comment]
@danrot change to develop.

**Comment by wachterjohannes** [comment]
with this PR we block a lot of "meaningfull" names like `400x400` or `170x170` which are the sulu internal formats. they also can not be reused in the application because they cannot be cropped. i would rename our internal formats in the same step and postfix (or prefix) them with sulu, like `170x170-sulu`

**Comment by danrot** [comment]
@wachterjohannes I already thought about that, but I wanted to do it in a separate PR... However, it would probably make more sense to already include it. Will do that.

**Comment by danrot** [comment]
Will do 50x50 later, have seen that there are even references in husky for that one...

File: src/Sulu/Bundle/MediaBundle/DependencyInjection/AbstractImageFormatCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+use Sulu\Bundle\MediaBundle\Media\FormatLoader\XmlFormatLoader10;
+use Sulu\Bundle\MediaBundle\Media\FormatLoader\XmlFormatLoader11;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\Config\Loader\DelegatingLoader;
+use Symfony\Component\Config\Loader\LoaderResolver;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
```
Comment (chirimoya) [inline src/Sulu/Bundle/MediaBundle/DependencyInjection/AbstractImageFormatCompilerPass.php]: phpdoc

File: src/Sulu/Bundle/MediaBundle/DependencyInjection/AbstractImageFormatCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+abstract class AbstractImageFormatCompilerPass implements CompilerPassInterface
+{
+    private $globalOptions;
```
Comment (chirimoya) [inline src/Sulu/Bundle/MediaBundle/DependencyInjection/AbstractImageFormatCompilerPass.php]: phpdoc

File: CHANGELOG.md [hunk 3]
```diff
@@ -76,6 +76,7 @@ CHANGELOG for Sulu
 
 * 1.4.2 (2016-11-24)
     * HOTFIX      #3032 [ContentBundle]         Fixed publishing for shadow page targeting drafts
+    * BUGFIX      #3040 [MediaBundle]           Throw exception when multiple formats have the same key
```
Comment (wachterjohannes) [inline CHANGELOG.md]: should be under `dev-develop`

**Comment by wachterjohannes** [comment]
add to upgrade that new formats was added and old will not be deleted.

**Comment by danrot** [comment]
@wachterjohannes Added the upgrade and fixed the readme. Should be fine now.

**Pull request merged**